lib/rspec/support/spec/stderr_splitter.rb

```diff
@@ -28,30 +26,12 @@ def reopen(*args)
         @orig_stderr.reopen(*args)
       end
 
-      # To work around JRuby error:
-      # can't convert RSpec::Support::StdErrSplitter into String
-      def to_io
-        @orig_stderr.to_io
-      end
-
-      # To work around JRuby error:
-      # TypeError: $stderr must have write method, RSpec::StdErrSplitter given
       def write(line)
-        return if line =~ %r{^\S+/gems/\S+:\d+: warning:} # http://rubular.com/r/kqeUIZOfPG
-
-        # Ruby 2.7.0 warnings from keyword argments span multiple lines, extend check above
-        # to look for the next line.
-        return if @last_line =~ %r{^\S+/gems/\S+:\d+: warning:} &&
-                  line =~ %r{warning: The called method .* is defined here}
-
-        # Ruby 2.7.0 complains about hashes used in place of keyword arguments
-        # Aruba 0.14.2 uses this internally triggering that here
-        return if line =~ %r{lib/ruby/2\.7\.0/fileutils\.rb:622: warning:}
+        # Ignore warnings coming from gems, specifically Rails on Ruby 2.5+
+        return if line =~ %r{^\S+/gems/\S+:\d+: warning:}
```

Comment:
Don't change this logic when its unrelated, we don't want to become less specific.